src/AppDataTrail/DataTrail.tsx

```diff
@@ -114,6 +114,12 @@ export class DataTrail extends SceneObjectBase<DataTrailState> implements SceneO
   }
 
   updateFromUrl(values: SceneObjectUrlValues) {
+    if (this.state.embedded) {
+      // In embedded mode, we want to avoid clearing a metric from the trail state
+      // when the trail has been freshly instantiated and the URL doesn't yet contain the metric.
+      return;
+    }
```

Comment:
It used to work this way, but there was a regression in https://github.com/grafana/metrics-drilldown/pull/670/files#diff-a9add872af423d0c1e1f9ff70772cf38ff8e4bb3ddfabebfbbc067bbfce53f9d